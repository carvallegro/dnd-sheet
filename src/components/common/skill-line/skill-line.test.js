import React from 'react'
import { shallow } from 'enzyme'

import { SkillLine, SkillGrid } from './index'

describe('<SkillLine />', () => {
  it('renders', () => {
    const wrapper = shallow(<SkillLine />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with props', () => {
    const wrapper = shallow(
      <SkillLine
        value={10}
        name={'Test Skill'}
        attribute={'TST'}
        isProficientw
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('<SkillGrid />', () => {
  it('renders', () => {
    const wrapper = shallow(<SkillGrid />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with rows', () => {
    const wrapper = shallow(<SkillGrid rows={2} />)
    expect(wrapper).toMatchSnapshot()
  })
})
