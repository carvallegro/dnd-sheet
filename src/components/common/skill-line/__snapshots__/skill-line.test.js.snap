// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkillGrid /> renders 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: 0.2rem;
  grid-auto-flow: column;
  grid-template-rows: repeat(",
          [Function],
          ", 1fr);
  grid-auto-columns: auto;
",
        ],
      },
      "defaultProps": Object {
        "rows": 6,
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "rows": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  rows={6}
/>
`;

exports[`<SkillGrid /> renders with rows 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": false,
        "rules": Array [
          "
  display: grid;
  grid-gap: 0.2rem;
  grid-auto-flow: column;
  grid-template-rows: repeat(",
          [Function],
          ", 1fr);
  grid-auto-columns: auto;
",
        ],
      },
      "defaultProps": Object {
        "rows": 6,
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "rows": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  rows={2}
/>
`;

exports[`<SkillLine /> renders 1`] = `
<styled.div>
  <Checkbox
    name="skill_undefined"
    onChange={[Function]}
    size="xsmall"
  />
  <styled.p />
  <styled.p />
  <Styled(styled.p) />
</styled.div>
`;

exports[`<SkillLine /> renders with props 1`] = `
<styled.div>
  <Checkbox
    name="skill_test_skill"
    onChange={[Function]}
    size="xsmall"
  />
  <styled.p>
    10
  </styled.p>
  <styled.p>
    Test Skill
  </styled.p>
  <Styled(styled.p)>
    TST
  </Styled(styled.p)>
</styled.div>
`;
